server/src/main/java/io/crate/analyze/AnalyzedTableElements.java

```diff
@@ -418,28 +440,73 @@ private static void validateAndFormatExpression(Symbol function,
+                                          boolean isAddColumn) {
 
         DataType<?> type = columnDefinition.dataType() == null ? DataTypes.UNDEFINED : columnDefinition.dataType();
         DataType<?> realType = ArrayType.NAME.equals(columnDefinition.collectionType())
             ? new ArrayType<>(type)
             : type;
 
-        SimpleReference simpleRef = new SimpleReference(
-            new ReferenceIdent(relationName, columnDefinition.ident()),
-            RowGranularity.DOC,
-            realType,
-            columnDefinition.position,
-            null // not required in this context
-        );
-        Reference ref = columnDefinition.isGenerated()
-            ? new GeneratedReference(simpleRef, columnDefinition.formattedGeneratedExpression(), null)
-            : simpleRef;
+        Reference ref;
+        if (type.id() == GeoShapeType.ID) {
+            Map<String, Object> geoMap = new HashMap<>();
+            if (isAddColumn && columnDefinition.geoProperties() != null) {
+                // applySettings validates geo properties.
+                // If this method called from CreateTablePlan geoProperties are not yet resolved, they are still Literals. No need to validate, it will be done later.
+                // In case of ADD COLUMN we have to validate geo properties.
+                GeoSettingsApplier.applySettings(geoMap, (GenericProperties<Object>) columnDefinition.geoProperties(), columnDefinition.geoTree());
+            }
+            Float distError = (Float) geoMap.get("distance_error_pct");
+            ref = new GeoReference(
+                columnDefinition.position,
+                new ReferenceIdent(relationName, columnDefinition.ident()),
+                columnDefinition.hasNotNullConstraint(),
+                columnDefinition.geoTree(),
+                (String) geoMap.get("precision"),
+                (Integer) geoMap.get("tree_levels"),
+                distError != null ? distError.doubleValue() : null
+            );
+        } else if (isAddColumn && columnDefinition.analyzer() != null) {
+            // We are sending IndexReference since it's the only Reference implementation having 'analyzer'.
+            // However, IndexReference is dedicated to reflect index declaration like 'INDEX some_index using fulltext(some_col) with (analyzer = 'english')'
+            // Hence, we ignore copyTo which is irrelevant for ADD COLUMN.
+            // columnDefinition.isIndexColumn() cannot be used here, it's always false for ADD COLUMN.
+            ref = new IndexReference(
+                columnDefinition.position,
+                columnDefinition.hasNotNullConstraint(),
+                !columnDefinition.docValuesSpecifiedAndDisabled(),
+                new ReferenceIdent(relationName, columnDefinition.ident()),
+                columnDefinition.indexConstraint(),
+                List.of(), //copyTo is irrelevant for ADD COLUMN
+                columnDefinition.analyzer()
```

Comment:
I'm re-using existing code to convert `AnalyzedColumnDefinition` into `List<Reference>`. Noticed that 
Geo and Index reference were not handled here - not sure about consequences of that, couldn't find any problematic scenario, AFAIK before it was used only on Create Analysis stage for subscripts. (see also https://crate.slack.com/archives/CBPLN6WJK/p1667900555817769). 

difference with master only that GeoReference is handled now - which I think is fine, as it's basically `SimpleReference` enriched with her properties, so old code will not be broken

Decided not to add `IndexReference` for `CREATE TABLE`, 
first pushed https://github.com/crate/crate/pull/13238/commits/b0835315f4cf061846b063bc912aa83b1aad70c2 had failures like cannot cast Literal to String because analyzer was not yet resolved and added fallback to old scenario in https://github.com/crate/crate/pull/13238/commits/a1fe53d75e0a9672605fa9ebe0117f06f7197764

Reply:
Sorry I don't quite follow. Is this needed for the PR or not?

Reply:
Yes, we need to adapt reused method as we cannot take it as is. 

The change has **almost** no affect on existing usage (old usage  is when `isAddColumn == false`) - for `IndexReference` we fall back to old behavior, for GeoReference we return Geo instead of Simple, which is basically same thing just enriched with geo properties, does not harm existing usage in `CREATE TABLE` analyzer.

Reply:
upd: now reused method falls back to old behavior if used NOT for `ADD COLUMN` - https://github.com/crate/crate/pull/13238/commits/4664628f7ea3b01b13d8cf054d59306e81a6bfbb

Initial idea was that we have to handle all cases when this method is used in CREATE TABLE (couldn't find a bug but thought would silently fix if there is any) - but let's keep the scope smaller